### PR TITLE
chore(api): clean up public api for platform-browser/testing

### PR DESCRIPTION
**Pull request by juliemr**
Mostly, removing things that were never intended to be exported publicy.

### CODE REVIEW

**Comment by vikerman** [comment]
LGTM

File: modules/@angular/platform-server/testing/server.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import {MockLocationStrategy} from '@angular/common/testing';
 import {COMPILER_PROVIDERS, DirectiveResolver, ViewResolver, XHR} from '@angular/compiler';
 import {MockDirectiveResolver, MockViewResolver, TestComponentBuilder, TestComponentRenderer} from '@angular/compiler/testing';
 import {APPLICATION_COMMON_PROVIDERS, APP_ID, NgZone, PLATFORM_COMMON_PROVIDERS, PLATFORM_INITIALIZER, RootRenderer} from '@angular/core';
-import {BrowserDetection} from '@angular/platform-browser/testing';
+// import {BrowserDetection} from '@angular/platform-browser/testing/browser_util';
```
Comment (IgorMinar) [inline modules/@angular/platform-server/testing/server.ts]: remove

File: modules/@angular/platform-server/testing/server.ts [hunk 2]
```diff
@@ -15,7 +15,7 @@ import {Log} from '@angular/core/testing';
 
 function initServerTests() {
   Parse5DomAdapter.makeCurrent();
-  BrowserDetection.setup();
+  // BrowserDetection.setup();
```
Comment (IgorMinar) [inline modules/@angular/platform-server/testing/server.ts]: remove

File: tools/public_api_guard/public_api_spec.ts [hunk 3]
```diff
@@ -1606,37 +1606,8 @@ const PLATFORM_BROWSER = [
...
-  'NgMatchers.toHaveCssClass(expected:any):boolean',
-  'NgMatchers.toHaveCssStyle(expected:any):boolean',
-  'NgMatchers.toHaveText(expected:any):boolean',
-  'NgMatchers.toImplement(expected:any):boolean',
-  'NgMatchers.toMatchPattern(expectedMessage:any):boolean',
-  'NgMatchers.toThrowErrorWith(expectedMessage:any):boolean',
-  'normalizeCSS(css:string):string',
-  'stringifyElement(el:any):string',
-  'var browserDetection:BrowserDetection',
-  'var expect:(actual: any) => NgMatchers',
```
Comment (IgorMinar) [inline tools/public_api_guard/public_api_spec.ts]: I think we need to document this one as a breaking change incase people were importing these helpers from our testing packages.

please see: https://github.com/angular/angular/commit/e1e5c40ef7b8f0219658577589c857e0e1d960d7

also the commit message type should be "fix" and not "chore", chore is only for non-source code changes. see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#type

**Comment by IgorMinar** [comment]
otherwise lgtm after rebase

File: modules/@angular/platform-server/testing/server.ts (hunk 1 above)
Comment (juliemr) [inline modules/@angular/platform-server/testing/server.ts]: done.

File: modules/@angular/platform-server/testing/server.ts (hunk 2 above)
Comment (juliemr) [inline modules/@angular/platform-server/testing/server.ts]: done.

**Comment by juliemr** [comment]
done, waiting for Travis to be green.

**Comment by tycho01** [comment]
I was using `expect(el).toHaveText(..)` with this public, but now managed to replace it with `expect(el.textContent).toEqual(...)`.

**Pull request merged**